### DISCUSSION TITLE
How do you handle `useFormState`?

### DISCUSSION THREAD

**Discussion by benjick** [next-safe-action/next-safe-action#66] [Q&A] [answered]
Hello!

How do you handle `useFormState` with `next-safe-action`? The function signatures does not match.

I've created this higher-order function to solve it:

```ts
function withFormState<T, R>(
  func: (inputs: T) => R,
): (arg0: unknown, inputs: T) => R {
  return (_, inputs: T) => func(inputs);
}

const myAction = withFormState(action(schema, async (data) => {
  ...
}));
```

Is this fine? Or is there another preferred way of solving this?

**Comment by TheEdoRan** [comment, marked as answer]
Hi, Form Actions support will land in v7, I'm still looking for a way to properly implement them, but your snippet will surely help development. Please track #29 for updates, you can contribute to the discussion there too. Thank you!

**Reply by benjick** [reply]
I totally missed that issue, thank you!
